examples/gbasf2_fei_training/fei_grid_workflow.py

```diff
@@ -0,0 +1,473 @@
+        # which is (ab)used to create that site list
+        yield FEITrainingTask(
+            mode="Training",
+            stage=-1,
+        )
+
+        # need .xml training files from current stage of FEI training
+        if self.stage > -1:
+            yield FEITrainingTask(
+                mode="Training",
+                stage=self.stage,
+            )
+
+        # need .xml training files from all previous stages of FEI training,
+        # beginning with stage 1
+        if self.stage > 0:
+
+            for fei_stage in range(self.stage):
+
+                yield FEITrainingTask(
+                    mode="Training",
+                    stage=fei_stage,
+                )
+
+    def run(self):
+
+        # create tarball with all required input files for FEIAnalysisTask
+        outputs = [outs[0] for outs in self.get_input_file_names().values()
+                   if outs[0].endswith('.root') or outs[0].endswith('.xml')]
+        taroutname = self.get_output_file_name("fei_analysis_inputs.tar.gz")
+        taroutdir = os.path.dirname(taroutname)
+        outputsstring = ' '.join(['"'+o+'"' for o in outputs])
+        tarcmd = f"rm -f {self.get_output_file_name('successfull_input_upload.txt')}; "
+        tarcmd += f"cp {outputsstring} {taroutdir}; "
+        tarcmd += f"pushd {taroutdir}; "
+        baseoutputsstring = ' '.join(['"'+os.path.basename(o)+'"' for o in outputs])
+        tarcmd += f"tar -vczf {taroutname} {baseoutputsstring}; "
+        tarcmd += f"rm -f {baseoutputsstring}; popd"
+        os.system(tarcmd)
+
+        # upload tarball to initial storage element
+        timestamp = datetime.datetime.now().strftime("_%b-%d-%Y_%H-%M-%S")
+        foldername = os.path.join(self.remote_tmp_directory.rstrip('/')+timestamp, "stage"+str(self.stage))
+        completed_copy = run_with_gbasf2(shlex.split(f"gb2_ds_put -d {self.remote_initial_se} "
+                                         f"-i {taroutdir} --datablock sub00 {foldername}"))
+
+        # replicate tarball to other storage element sites (defined by used input datasets)
+        dataset_sites = [site.strip() for site in open(self.get_input_file_names('dataset_sites.txt')[0], 'r').readlines()]
+        completed_replicas = []
+        for ds_site in dataset_sites:
+            completed_replicas.append(run_with_gbasf2(shlex.split(f"gb2_ds_rep {foldername}/"
+                                      f"sub00 -d {ds_site} -s {self.remote_initial_se} --force")))
+
+        if sum([proc.returncode for proc in completed_replicas + [completed_copy]]) == 0:
+            with open(f"{self.get_output_file_name('successfull_input_upload.txt')}", "w") as timestampfile:
+                timestampfile.write(timestamp)
+                timestampfile.close()
+
+
+class ProduceStatisticsTask(luigi.WrapperTask):
+
+    def requires(self):
+
+        # yield FEITrainingTask(
+        #     mode="Training",
+        #     stage=0,
+        # )
```

Comment:
I'm not sure why large chunks of code are commented out but when you finalize this PR, all comments should serve documentation purposes, either documenting a part of code or showing an example for an alternative. The examples are long anyway

Reply:
I'd prefer to keep these example, how the workflow can be started to have them all at hand.

The main one would be the one currently activated, but for testing purposes, the ones in the comments can be used.

Reply:
:+1: This is a valid reason as it roughly falls into the documentation purposes, as it shows what else could be used.